### PR TITLE
Performance improvements for page copy

### PR DESCRIPTION
**Pull request by kaedroho**
See commit log for details.

### CODE REVIEW

**Review by jacobtoppm — APPROVED** [review]
Looks good to me, aside from one the one comment! Do you have any stats on the performance improvement?

**Comment by kaedroho** [comment]
> Do you have any stats on the performance improvement?

I did but had to roll back one of the changes so this might not be accurate anymore:

To copy the entire bakerydemo, this reduced resource usage as follows:

Time: ~22s => ~9s
Queries: 1600 => 550

**Comment by jacobtoppm** [comment]
> > Do you have any stats on the performance improvement?
> 
> I did but had to roll back one of the changes so this might not be accurate anymore:
> 
> To copy the entire bakerydemo, this reduced resource usage as follows:
> 
> Time: ~22s => ~9s
> Queries: 1600 => 550

Nice! That's pretty substantial

File: wagtail/core/models.py [hunk 1]
```diff
@@ -1177,7 +1177,7 @@ def get_admin_display_title(self):
         return self.draft_title or self.title
 
     def save_revision(self, user=None, submitted_for_moderation=False, approved_go_live_at=None, changed=True,
-                      log_action=False, previous_revision=None):
+                      log_action=False, previous_revision=None, clean=True):
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/core/models.py:1180]: Done!

**Comment by kaedroho** [comment]
@jacobtoppm I've been able to reinstate another performance improvement in the last commit, so the stats above should be accurate!

Please could you review the latest commit?

**Review by jacobtoppm — APPROVED** [review]
This looks good to me! Nice work

**Pull request merged**